test/libsyncengine/propagation/executor/testexecutorworker.cpp

```diff
@@ -180,4 +162,42 @@ void TestExecutor::testCheckLiteSyncInfoForCreate() {
 #endif
 }
 
+void TestExecutorWorker::testFixModificationDate() {
+    // Create temp directory
+    const LocalTemporaryDirectory temporaryDirectory;
+    // Create file
+    const SyncName filename = Str("test_file.txt");
+    const SyncPath path = temporaryDirectory.path / filename;
+    {
+        std::ofstream ofs(path);
+        ofs << "abc";
+        ofs.close();
+    }
+
+    // Update DB
+    DbNode dbNode(0, 123, filename, filename, "lid", "rid", defaultTime, defaultTime, defaultTime, NodeTypeFile, defaultSize,
+                  "cs");
+    DbNodeId dbNodeId;
+    bool constraintError = false;
+    _syncPal->_syncDb->insertNode(dbNode, dbNodeId, constraintError);
+
+    // Generate sync operation
+    std::shared_ptr<Node> node =
+        std::make_shared<Node>(std::nullopt, ReplicaSideLocal, filename, NodeTypeFile, "lid", defaultTime, 12345, defaultSize);
+    std::shared_ptr<Node> correspondingNode = std::make_shared<Node>(std::nullopt, ReplicaSideLocal, filename, NodeTypeFile,
+                                                                     "rid", defaultTime, defaultTime, defaultSize);
+    SyncOpPtr op = std::make_shared<SyncOperation>();
+    op->setAffectedNode(node);
+    op->setCorrespondingNode(correspondingNode);
+
+    _syncPal->_executorWorker->fixModificationDate(op, path);
```

Comment:
Public accessor?

Reply:
Not sure if it is a good idea to have a public accessor on `_executorWorker `. We do not want any other module to be able to access and modify the `Executor` (except from the test class).
What do you think?
IMHO, we definitively put to too much friend classes, but it is ok to define the test class as a friend of the base class (e.g. : `TestExecutorWorker` is a friend of `ExecutorWorker`). However, the SyncPal class has too many friends, and each unit test class should depend only on the class it is testing (e.g. : `TestExecutorWorker` should not have to allocate a SyncPal variable). But such a big change should be the topic of a dedicated PR